packages/stryker-api/src/core/File.ts

```diff
@@ -1,29 +1,55 @@
-export enum FileKind {
-  'Binary',
-  'Text',
-  'Web',
-}
 
-export type File = BinaryFile | TextFile | WebFile;
+/** 
+ * Represents a file within Stryker. Could be a strictly in-memory file.
+*/
+export default class File {
 
-export interface FileDescriptor {
-  name: string;
-  included: boolean;
-  mutated: boolean;
-  transpiled: boolean;
-  kind: FileKind;
-}
+  private _textContent: string | undefined;
+  private _content: Buffer;
 
-export interface TextFile extends FileDescriptor {
-  content: string;
-  kind: FileKind.Text;
-}
+  /**
+   * Creates a new File to be used within Stryker.
+   * @param name The full name of the file (inc path)
+   * @param content The buffered or string content of the file
+   */
+  constructor(public readonly name: string, content: Buffer | string) {
+    if (typeof content === 'string') {
+      this.textContent = content;
+    } else {
+      this.content = content;
+    }
+  }
+
+  /**
+   * Gets the underlying content as buffer.
+   */
+  get content(): Buffer {
+    return this._content;
+  }
 
-export interface WebFile extends FileDescriptor {
-  kind: FileKind.Web;
- }
- 
-export interface BinaryFile extends FileDescriptor {
-  content: Buffer;
-  kind: FileKind.Binary;
-}
\ No newline at end of file
+  /**
+   * Sets the underlying content as buffer.
+   */
+  set content(value: Buffer) {
```

Comment:
File class should become immutable. Also applies for the `set textContent` function.